integrationTests/singleShard/block/executingMiniblocksSc_test.go

```diff
@@ -148,57 +232,396 @@ func TestProcessesJoinGameOf100PlayersRewardAndEndgame(t *testing.T) {
+	nodes[0] = integrationTests.NewTestProcessorNode(maxShards, 0, 0, advertiserAddr)
+	nodes[1] = integrationTests.NewTestProcessorNode(maxShards, 0, 0, advertiserAddr)
+	nodes[2] = integrationTests.NewTestProcessorNode(maxShards, 1, 0, advertiserAddr)
+	nodes[3] = integrationTests.NewTestProcessorNode(maxShards, 1, 0, advertiserAddr)
+	nodes[4] = integrationTests.NewTestProcessorNode(maxShards, sharding.MetachainShardId, 0, advertiserAddr)
+	nodes[5] = integrationTests.NewTestProcessorNode(maxShards, sharding.MetachainShardId, 0, advertiserAddr)
+
+	idxProposer := 0
+	nrPlayers := 100
+	players := make([]*integrationTests.TestWalletAccount, nrPlayers)
+	for i := 1; i < nrPlayers; i++ {
+		players[i] = integrationTests.CreateTestWalletAccount(nodes[idxProposer].ShardCoordinator, 0)
+	}
+
+	defer func() {
+		_ = advertiser.Close()
+		for _, n := range nodes {
+			_ = n.Messenger.Close()
+		}
+	}()
+
+	for _, n := range nodes {
+		_ = n.Messenger.Bootstrap()
+	}
+
+	fmt.Println("Delaying for nodes p2p bootstrap...")
+	time.Sleep(time.Second)
+
+	round := uint64(0)
+	round = incrementAndPrintRound(round)
+
+	hardCodedSk, _ := hex.DecodeString("5561d28b0d89fa425bbbf9e49a018b5d1e4a462c03d2efce60faf9ddece2af06")
+	hardCodedScResultingAddress, _ := hex.DecodeString("000000000000000000005fed9c659422cd8429ce92f8973bba2a9fb51e0eb3a1")
+	nodes[idxProposer].LoadTxSignSkBytes(hardCodedSk)
+
+	initialVal := big.NewInt(10000000)
+	topUpValue := big.NewInt(500)
+	integrationTests.MintAllNodes(nodes, initialVal)
+	integrationTests.MintAllPlayers(nodes, players, initialVal)
+
+	nrRoundsToPropagateMultiShard := 5
+	if maxShards == 1 {
+		nrRoundsToPropagateMultiShard = 1
+	}
+
+	idxProposers := make([]int, 3)
+	idxProposers[0] = 0
+	idxProposers[1] = 2
+	idxProposers[2] = 4
+
+	for i := 0; i < nrRoundsToPropagateMultiShard; i++ {
+		time.Sleep(time.Second)
+		deployScTx(nodes, idxProposer, string(scCode))
+		time.Sleep(time.Second)
+		proposeBlock(nodes, idxProposers, round)
+		time.Sleep(time.Second)
+		syncBlock(t, nodes, idxProposers, round)
 		round = incrementAndPrintRound(round)
+	}
+
+	runMultipleRoundsOfTheGame(
+		t,
+		100,
+		nrPlayers,
+		nodes,
+		players,
+		topUpValue,
+		hardCodedScResultingAddress,
+		round,
+		maxShards,
+		idxProposers,
+	)
+
+	checkRootHashes(t, nodes, []int{0})
+
+	time.Sleep(time.Second)
+}
+
+func getPercentageOfValue(value *big.Int, percentage float64) *big.Int {
+	x := new(big.Float).SetInt(value)
+	y := big.NewFloat(percentage)
+
+	z := new(big.Float).Mul(x, y)
+
+	op := big.NewInt(0)
+	result, _ := z.Int(op)
+
+	return result
+}
+
+func runMultipleRoundsOfTheGame(
+	t *testing.T,
+	nrRounds, nrPlayers int,
+	nodes []*integrationTests.TestProcessorNode,
+	players []*integrationTests.TestWalletAccount,
+	topUpValue *big.Int,
+	hardCodedScResultingAddress []byte,
+	round uint64,
+	nrShards uint32,
+	idxProposers []int,
+) {
+	rMonitor := &statistics.ResourceMonitor{}
+	nrRewardedPlayers := 10
```

Comment:
nr again, please use num in all occurrences

Reply:
done